corehq/apps/users/tests/test_util.py

```diff
@@ -187,3 +190,84 @@ def test_user_is_not_deactivated_and_no_logs(self):
         self.assertFalse(
             UserHistory.objects.filter(user_id=self.inactive_user.user_id).exists()
         )
+
+
+class TestGenerateMobileUsername(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain = 'test-domain'
+        cls.domain_obj = create_domain('test-domain')
+        cls.addClassCleanup(cls.domain_obj.delete)
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.user = CommCareUser.create(self.domain, 'test-user@test-domain.commcarehq.org', 'abc123', None, None)
+        self.addCleanup(self.user.delete, self.domain, None)
+
+    def test_successfully_generated_username(self):
+        try:
+            username = generate_mobile_username('test-user-1', self.domain)
+        except ValidationError:
+            self.fail(f'Unexpected raised exception: {ValidationError}')
+
+        self.assertEqual(username, 'test-user-1@test-domain.commcarehq.org')
+
+    def test_invalid_username_double_period_message(self):
+        try:
+            generate_mobile_username('test..user', self.domain)
+        except ValidationError as e:
+            self.assertEqual(e.message, 'Username may not contain consecutive . (period).')
+        else:
+            self.fail(f'Expected raised exception: {ValidationError}')
```

Comment:
nit: The usual way to validate exception handling is as follows:
```suggestion
        with self.assertRaises(ValidationError) as cm:
            generate_mobile_username('test..user', self.domain)
        self.assertEqual(cm.exception.message, 'Username may not contain consecutive . (period).')
```

Reply:
See e9bb80ded2dd47c3a2f1c608fe2545524ed91b59